packages/contentstack-auth/src/commands/auth/login.ts

```diff
@@ -21,42 +29,61 @@ export default class LoginCommand extends Command {
       description: 'User name',
       multiple: false,
       required: false,
+      exclusive: ['oauth'],
     }),
     password: flags.string({
       char: 'p',
       description: 'Password',
       multiple: false,
       required: false,
+      exclusive: ['oauth'],
+    }),
+    oauth: flags.boolean({
+      description: 'Boolean flag for OAuth login to contentstack',
+      required: false,
+      default: false,
+      exclusive: ['username', 'password'],
+      hidden: true,
     }),
   };
 
   static aliases = ['login'];
 
   async run(): Promise<any> {
-    const managementAPIClient = await managementSDKClient({host: this.cmaHost})
-    const { flags: loginFlags } = await this.parse(LoginCommand)
+    const managementAPIClient = await managementSDKClient({ host: this.cmaHost });
+    const { flags: loginFlags } = await this.parse(LoginCommand);
     authHandler.client = managementAPIClient;
 
     try {
-      const username = loginFlags?.username || (await interactive.askUsername());
-      const password = loginFlags?.password || (await interactive.askPassword());
-      logger.debug('username', username);
-      await this.login(username, password);
+      const oauth = loginFlags?.oauth;
+      if (oauth === true) {
+        oauthHandler.host = this.cmaHost;
+        await oauthHandler.oauth();
+      } else {
+        const username = loginFlags?.username || (await interactive.askUsername());
+        const password = loginFlags?.password || (await interactive.askPassword());
+        logger.debug('username', username);
+        await this.login(username, password);
+      }
     } catch (error) {
-      logger.error('login  failed', error.message);
+      logger.error('login failed', error.message);
       cliux.print('CLI_AUTH_LOGIN_FAILED', { color: 'yellow' });
       cliux.print(error.message.message ? error.message.message : error.message, { color: 'red' });
     }
   }
 
   async login(username: string, password: string): Promise<void> {
-    const user: User = await authHandler.login(username, password);
-    if (typeof user !== 'object' || !user.authtoken || !user.email) {
-      throw new CLIError('Failed to login - invalid response');
+    try {
+      const user: User = await authHandler.login(username, password);
+      if (typeof user !== 'object' || !user.authtoken || !user.email) {
+        throw new CLIError('Failed to login - invalid response');
+      }
+      await oauthHandler.setConfigData('basicAuth', user);
+      logger.info('successfully logged in');
+      cliux.success('CLI_AUTH_LOGIN_SUCCESS');
+    } catch (error) {
+      console.error(error);
+      throw error;
```

Comment:
Suggestion: Do we really need to throw the error and catch it again and throw? Can we just stop the process? Because if authentication failed user can't use CLI right?